### PR TITLE
Basic documentation update for engine branch

### PR DESCRIPTION
**Pull request by thrasher-**
# Description

Basic documentation update for gRPC/gctcli, root readme and a basic unified API doc with code examples.

## Type of change

Documentation update

# How Has This Been Tested?

Clicking on ALL THE LINKS

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool 
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally and on Travis with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/EXCHANGE_API.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+A full breakdown of all the supported wrapper funcs can be found [here.](https://github.com/thrasher-corp/gocryptotrader/blob/engine/exchanges/interfaces.go#L16)
+Please note that these change on a regular basis as more features get added on
+a regular basis.
+
+Each exchange supports public API endpoints which don't require any authentication
+(fetching ticker, orderbook, trade data) and also private API endpoints (which
+require authentication). Some examples include submitting, canceling and fetching
+open orders). To use the authenticated API endpoints, you'll need to set your API
+credentials in either the `config.json` file or when you initialise an exchange in
+your application, and also have the approriate key permissions set for the exchange.
```
Comment (gloriousCode) [inline docs/EXCHANGE_API.md]: appropriate

File: gctrpc/README.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+* `protoc-gen-grpc-gateway`
+* `protoc-gen-swagger`
+* `protoc-gen-go`
+
+Make sure that your `$GOBIN` is in your `$PATH`.
+
+## Usage
+
+After the above depenancies are required, make necessary changes to the `rpc.proto`
```
Comment (gloriousCode) [inline gctrpc/README.md]: dependencies

File: docs/EXCHANGE_API.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+has its own wrapper file which maps the exchanges own RESTful endpoints into a
+standardised way which allows an easy way for the bot and developers to work with.
+
+A full breakdown of all the supported wrapper funcs can be found [here.](https://github.com/thrasher-corp/gocryptotrader/blob/engine/exchanges/interfaces.go#L16)
+Please note that these change on a regular basis as more features get added on
+a regular basis.
+
+Each exchange supports public API endpoints which don't require any authentication
+(fetching ticker, orderbook, trade data) and also private API endpoints (which
+require authentication). Some examples include submitting, canceling and fetching
```
Comment (gloriousCode) [inline docs/EXCHANGE_API.md]: cancelling

File: docs/README.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# GoCryptoTrader Documentation
```
Comment (gloriousCode) [inline docs/README.md]: These readmes are lacking the following underneath the h1s
```
<img src="https://github.com/thrasher-corp/gocryptotrader/blob/master/web/src/assets/page-logo.png?raw=true" width="350px" height="350px" hspace="70">


[![Build Status](https://travis-ci.org/thrasher-corp/gocryptotrader.svg?branch=master)](https://travis-ci.org/thrasher-corp/gocryptotrader)
[![Software License](https://img.shields.io/badge/License-MIT-orange.svg?style=flat-square)](https://github.com/thrasher-corp/gocryptotrader/blob/master/LICENSE)
[![GoDoc](https://godoc.org/github.com/thrasher-corp/gocryptotrader?status.svg)](https://godoc.org/github.com/thrasher-corp/gocryptotrader/config)
[![Coverage Status](http://codecov.io/github/thrasher-corp/gocryptotrader/coverage.svg?branch=master)](http://codecov.io/github/thrasher-corp/gocryptotrader?branch=master)
[![Go Report Card](https://goreportcard.com/badge/github.com/thrasher-corp/gocryptotrader)](https://goreportcard.com/report/github.com/thrasher-corp/gocryptotrader)


This config package is part of the GoCryptoTrader codebase.

## This is still in active development

You can track ideas, planned features and what's in progresss on this Trello board: [https://trello.com/b/ZAhMhpOy/gocryptotrader](https://trello.com/b/ZAhMhpOy/gocryptotrader).

Join our slack to discuss all things related to GoCryptoTrader! [GoCryptoTrader Slack](https://join.slack.com/t/gocryptotrader/shared_invite/enQtNTQ5NDAxMjA2Mjc5LTc5ZDE1ZTNiOGM3ZGMyMmY1NTAxYWZhODE0MWM5N2JlZDk1NDU0YTViYzk4NTk3OTRiMDQzNGQ1YTc4YmRlMTk)

```
Is that okay?

Comment (thrasher-) [reply docs/README.md]: Yep!

File: cmd/gctcli/README.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+## Background
+
+GoCryptoTrader utilises gRPC for client/server interaction. Authentication is done
+by a self signed TLS cert, which only supports connections from localhost and also
+through basic authorisation specified by the users config file.
+
+## Usage
+
+GoCryptoTrader must be running with gRPC enabled in order to use  the client features.
```
Comment (gloriousCode) [inline cmd/gctcli/README.md]: There are TWO SPACES HERE - though readme renderers ignore it so you're fine.

**Pull request merged**